Services/Mail/classes/class.ilMailGUI.php

```diff
@@ -281,4 +281,29 @@ private function showHeader() : void
             $DIC->tabs()->setTabActive('fold');
         }
     }
+
+    /**
+     * Toggle explorer node
+     */
+    protected function toggleExplorerNodeState()
+    {
+        $nodeId = (int) ($this->httpRequest->getQueryParams()['node_id'] ?? 0);
+        $priorState = (int) ($this->httpRequest->getQueryParams()['prior_state'] ?? 0);
+
+        if ($nodeId > 0) {
+            $store = new ilSessionIStorage('expl2');
+
+            $openNodes = (array) unserialize($store->get('mail_tree'));
+            if (0 === $priorState && !in_array($nodeId, $openNodes)) {
+                $openNodes[] = $nodeId;
+                $store->set('mail_tree', serialize($openNodes));
+            } elseif (1 === $priorState && in_array($nodeId, $openNodes)) {
+                $key = array_search($nodeId, $openNodes);
+                unset($openNodes[$key]);
+                $store->set('mail_tree', serialize($openNodes));
+            }
+        }
+        exit();
```

Comment:
Consumer code needs to take care of toggle states of the tree?

Reply:
Currently we have no central mechanism and controller for this.

For most trees this (or something similar) could be done/implemented in `ilExplorerBaseGUI` or `ilTreeExplorerGUI`.